src/schema/v2/me/user_flags.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  GraphQLError,
+  GraphQLFieldConfig,
+  GraphQLScalarType,
+  Kind,
+  ObjectValueNode,
+  print,
+  ValueNode,
+} from "graphql"
+import Maybe from "graphql/tsutils/Maybe"
+import { camelCase } from "lodash"
+import { ResolverContext } from "types/graphql"
+
+export const UserFlagsType = new GraphQLScalarType({
+  name: "UserFlagsType",
+  description:
+    "The user's user_flags. Values of keys in user_flags can be anything and keys may not be known beforehand.",
+  serialize: (value) => value,
+  parseValue: (value) => value,
+  parseLiteral: (ast, variables) => {
+    if (ast.kind !== Kind.OBJECT) {
+      throw new TypeError(
+        `Object cannot represent non-object value: ${print(ast)}`
+      )
+    }
+
+    return parseObject("Object", ast, variables)
+  },
+})
+
+export const userFlagsResolver = async (root, options, loaders) => {
+  const { collectorProfileLoader } = loaders
+  if (!collectorProfileLoader) {
+    throw new Error("You need to be signed in to perform this action")
+  }
+
+  const otherFlags: { [key: string]: any } = {}
+
+  const result = await collectorProfileLoader(options)
+  const { bio, icon, other_relevant_positions } = result
+  const profileIsComplete =
+    root.profession && bio && icon && other_relevant_positions
+
+  if (!profileIsComplete) {
+    otherFlags.collector_profile_incomplete_at = new Date().toISOString()
+  }
```

Comment:
**suggestion:** I would suggest to move this part to a separate function to get the comuputed Flags.

```typescript
const computedFlags = getComputedFlags(...)
//
const getComputedFlags = (...) => {
const computedFlags = {}

computedFlags[collector_profile_incomplete_at] = computeProfileComletetion(...)

}
```
this will make it easier for us to add computed flags in the future